### DISCUSSION TITLE
Why not switch MSRV to 1.57?

### DISCUSSION THREAD

**Discussion by vabka** [tokio-rs/tokio#4876] [General] [answered]
1.57 released 7 month ago, so it can be accepted under
> Tokio will keep a rolling MSRV (minimum supported rust version) policy of at least 6 months. When increasing the MSRV, the new Rust version must have been released at least six months ago. The current MSRV is 1.49.0.

PS: Why am i asking: https://github.com/hyperium/http/issues/564

**Comment by Darksonn** [comment, marked as answer]
We only bump the MSRV if there's a sufficiently good reason to bump it.

**Reply by Pzixel** [reply]
Is using panic in const fns is a good sufficiently reason to bump it?

**Reply by Darksonn** [reply]
No. Having to do the array indexing trick is only a really minor inconvenience.

Besides, your link goes to a different library. Bumping Tokio's MSRV without even using it in Tokio really doesn't make any sense. The authors of `http` could bump their MSRV if they don't want to wait for Tokio. Alternatively, then they could also add `build.rs` version detection to use the `panic!` only on sufficiently new versions of rustc - that would give the benefits of improved error messages to most users of the library without bumping the MSRV.
